### PR TITLE
fix(ui5-table): remove circular dependency from table and row

### PR DESCRIPTION
**Pull request by DonkeyCo**
- remove cyclic import in TableRowBase of Table.ts
- add TableUtils.ts, which adds duck-typed instanceof checks for safe imports

Fixes: #9247

### CODE REVIEW

File: packages/main/src/TableCellBase.ts [hunk 1]
```diff
@@ -49,6 +49,10 @@ abstract class TableCellBase extends UI5Element {
 	getFocusDomRef() {
 		return this;
 	}
+
+	isTableCellBase() {
```
Comment (aborjinik) [inline packages/main/src/TableCellBase.ts:53]: isTableCell

Comment (DonkeyCo) [reply packages/main/src/TableCellBase.ts:53]: @nnaydenow linked me this PR: https://github.com/SAP/ui5-webcomponents/pull/8822 referencing point 4.

Seems like the main issue is scoping, when you have multiple different instances of the framework in your application, instanceof is not that reliable anymore?

**Review by ilhan007 — APPROVED** [review]

**Review by aborjinik — COMMENTED** [review]
How about setting an attribute in the onEnterDom of base classes and check this attribute to determine?
For the Table this should not be necessary since the framework does this anyway.

**Comment by DonkeyCo** [comment]
We can certainly try that. It would be different to how other components would handle this, but I think the Table structure and its components is anyways pretty different to the existing components.

I'll try to implement the change in this PR: https://github.com/SAP/ui5-webcomponents/pull/9205

**Pull request merged**